### PR TITLE
GED kubedog doc generation

### PR DESCRIPTION
**Pull request by garomonegro**
I have been wanting to add automatic documentation generation for the `docs/syntax.md` file. Today I took the time to do it.  As next steps we need to integrate this into the CI such that it fails if running `go generate` alters `syntax.md`. That would mean the contributor did not update the docs as part of the PR.

### CODE REVIEW

File: docs/syntax.md [hunk 1]
```diff
@@ -21,6 +25,12 @@ Below you will find the step syntax next to the name of the method it utilizes.
 - `<GK> [the] (deployment|hpa|horizontalpodautoscaler|service|pdb|poddisruptionbudget|sa|serviceaccount) <any-characters-except-(")> is in namespace <any-characters-except-(")>` kdt.KubeContext.ResourceInNamespace
 - `<GK> [I] scale [the] deployment <any-characters-except-(")> in namespace <any-characters-except-(")> to <digits>` kdt.KubeContext.ScaleDeployment
 - `<GK> [the] (clusterrole|clusterrolebinding) with name <any-characters-except-(")> should be found` kdt.KubeContext.ClusterRbacIsFound
+- `<GK> [I] (add|remove) ?<any-characters-except-(")> as trusted entity to iam role <any-characters-except-(")>` kdt.AwsContext.IamRoleTrust
```
Comment (garomonegro) [inline docs/syntax.md:28]: What is that `?` character for? i think @Kyle-Wong introduced it.

File: docs/syntax.md [hunk 2]
```diff
@@ -21,6 +25,12 @@ Below you will find the step syntax next to the name of the method it utilizes.
...
+- `<GK> [I] (add|remove) cluster shared iam role` kdt.AwsContext.ClusterSharedIamOperation
```
Comment (garomonegro) [inline docs/syntax.md:29]: I had moved this 2 steps under the "AWS Steps" tittle, why do we need them up here @Kyle-Wong?

File: generate/syntax/main.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+const (
+	sourceFilePath         = "kubedog.go"
+	destinationFilePath    = "docs/syntax.md"
+	newLine                = "\n"
+	actionIndicator        = "//syntax-generation"
+	actionDelimiter        = ":"
+	actionBegin            = "begin"
+	actionEnd              = "end"
+	actionTittle           = "tittle"
+	tittleBeginning        = "## "
```
Comment (danielhelfand) [inline generate/syntax/main.go:22]: Should this be title? 

```suggestion
	titleBeginning        = "## "
```

File: generate/syntax/main.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	processedStepBeginning = "- "
+	stepIndicator          = "kdt.scenarioContext.Step"
+	stepDelimiter          = "`"
+	stepPrefix             = "^"
+	stepSuffix             = "$"
+	methodPrefix           = ","
+	methodSuffix           = ")"
+	markdownCodeDelimiter  = "`"
+	gherkinKeyword         = "<GK>"
+
```
Comment (danielhelfand) [inline generate/syntax/main.go:32]: Should we remove this extra space?

**Review by danielhelfand — COMMENTED** [review]
lgtm, just some minor comments

File: docs/syntax.md (hunk 1 above)
Comment (danielhelfand) [reply docs/syntax.md:28]: A question mark ( ? ) immediately following a character means match zero or one instance of this character. So the space before the name is optional I think? Not a regex pro by any means so could be wrong.

File: docs/syntax.md (hunk 2 above)
Comment (Kyle-Wong) [reply docs/syntax.md:29]: That might have been a mistake when I merged into my PR branch. The steps don’t need to be up here

File: docs/syntax.md (hunk 1 above)
Comment (Kyle-Wong) [reply docs/syntax.md:28]: That is a mistake, that parameter is not optional. Good catch

File: docs/syntax.md [hunk 5]
```diff
@@ -1,22 +1,56 @@
...
-9.	`<GK> [I] update [a] resource <filename> with <complete key> set to <value>` [UpdateResourceWithField](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.UpdateResourceWithField)
-10.	`<GK>  <number of> nodes with selector <complete key>=<value> should be (found|ready)` [NodesWithSelectorShouldBe](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.NodesWithSelectorShouldBe)
-11. `<GK>  [the] <deployment|hpa|horizontalpodautoscaler|service|pdb|poddisruptionbudget> <resource_name> is in namespace <namespace-name>` [ResourceInNamespace](https://pkg.go.dev/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceInNamespace)
-12. `<GK>  [I] scale [the] deployment <filename> in namespace <namespace-name> to <replica(s)>` [ScaleDeployment](https://pkg.go.dev/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ScaleDeployment)
+## Generic steps
+- `<GK> [I] wait for <digits> (minutes|seconds)` common.WaitFor
+- `<GK> the <non-whitespace-characters> command is available` common.CommandExists
+- `<GK> I run the <non-whitespace-characters> command with the <any-characters-except-(")> args the command (fails|succeeds)` common.RunCommand
```
Comment (danielhelfand) [inline docs/syntax.md:7]: This is my mistake, but can we add `and` like below:

```suggestion
- `<GK> I run the <non-whitespace-characters> command with the <any-characters-except-(")> args and the command (fails|succeeds)` common.RunCommand
```

Comment (garomonegro) [reply docs/syntax.md:7]: This is autogenerated, the change needs to be made in the `kubedog.go` file

Comment (danielhelfand) [reply docs/syntax.md:7]: Right, would you mind changing it there so we don't need another pr?

File: generate/syntax/main.go (hunk 3 above)
Comment (garomonegro) [reply generate/syntax/main.go:22]: Thanks!

File: docs/syntax.md (hunk 5 above)
Comment (garomonegro) [reply docs/syntax.md]: Of course, I did already.

**Pull request merged**